apps/dashr-uber-rasterizer/app.R

```diff
@@ -0,0 +1,317 @@
+        y_range <- c(relayout$`yaxis.range[0]`, relayout$`yaxis.range[1]`)
+      }
+      
+      else {
+        x_range <- c(min(ridesDf[,"Lon"], na.rm=T), -72.5)
+        y_range <- c(39.9, max(ridesDf[,"Lat"], na.rm=T))
+      }
+    }
+    return(list(x_range, y_range))
+  }
+)
+
+# Callback to generate rasterization plot.
+
+app$callback(
+  output(id = 'rasterizer-output', property = 'figure'),
+  params = list(
+    input(id = 'store', property = 'data'),
+    input(id = 'cmap', property = 'value'),
+    input(id = 'background', property = 'value'),
+    input(id = 'reduc', property = 'value'),
+    input(id = 'scaling', property = 'value'),
+    input(id = 'point-size', property = 'value')
+  ),
+  update_graph <- function(data, cmap, background, reduc, scale, point_size) {
+    color <- if(cmap == "blue") {
+      c("lightblue", "darkblue")
+    } else if(cmap =="viridis") {
+      color <- viridis(256)
+    } else if (cmap =="plasma"){
+      color <- plasma(256)
+    } else {
+      eval(parse(text = cmap))
+    }
+    
+    if(background != "black") {
+      color <- rev(color)
+    }
+    
+    len_col <- length(color)
+    
+    colorscale <- lapply(0:len_col,
+                         function(i) {
+                           if(i == 0) {
+                             return(list(0, background))
+                           } else {
+                             return(list(i/len_col, color[i]))
+                           }
+                         }
+    )
+    
+    x_min <- data[[1]][[1]]
+    x_max <- data[[1]][[2]]
+    y_min <- data[[2]][[1]]
+    y_max <- data[[2]][[2]]
+    
+    filtered_df_lat <- ridesDf[ridesDf[, "Lat"] > y_min & ridesDf[, "Lat"] < y_max, ]    
+    filtered_df_lon <- filtered_df_lat[filtered_df_lat[,"Lon"] > x_min & filtered_df_lat[,"Lon"] < x_max, ]
+    
+    colorbar_title <- ifelse(scale == "log", "Log(No. of Rides)", "No. of Rides")
+    # plot_ly requires a data.frame
+    return(
+      plot_ly(as.data.frame(filtered_df_lon), x = ~Lon, y = ~Lat,
+              colorscale = colorscale,
+              colorbar = list(title = colorbar_title)) %>%
+        add_rasterly_heatmap(reduction_func = reduc, scaling = scale, size = b <- if(point_size == 0) NULL else {point_size}) %>%
+        layout(font = list(color = 'rgb(226, 239, 250)'),
+               margin = list(l = 1, r = 1, b = 1, t = 15, pad = 0, autoexpand = TRUE),
+               paper_bgcolor='rgb(38, 43, 61)',
+               plot_bgcolor='rgb(38, 43, 61)',
+               xaxis = list(title = "Longitude",
+                            constrain = "domain",
+                            scaleanchor = "y",
+                            scaleratio = cos(40.8*pi/180)),
+               yaxis = list(title = "Latitude",
+                            constrain = "domain"))
+    )
+  }
+)
+
+
+if(appName != "") {
+  app$run_server(host = "0.0.0.0", port = Sys.getenv('PORT', 8050))
+} else {
+  app$run_server(host = "127.0.0.1", 
```

Comment:
@HammadTheOne Here I would probably default to using `debug=TRUE` and setting the host and port, but maybe not enabling the other options. `viewer = TRUE` has changed since hot reloading support was initially available, and is now `use_viewer = TRUE`, but this feature won't be available until `dev` gets merged into `master`.

Reply:
Agreed, fixed.